iroh-sync/src/store.rs

```diff
@@ -172,35 +176,250 @@ pub trait Store: std::fmt::Debug + Clone + Send + Sync + 'static {
     fn get_sync_peers(&self, namespace: &NamespaceId) -> Result<Option<Self::PeersIter<'_>>>;
 }
 
-/// Filter a get query onto a namespace
-#[derive(Debug, Serialize, Deserialize)]
-pub enum GetFilter {
-    /// No filter, list all entries
-    All,
-    /// Filter for exact key match
-    Key(Vec<u8>),
-    /// Filter for key prefix
-    Prefix(Vec<u8>),
-    /// Filter by author
-    Author(AuthorId),
-    /// Filter by key prefix and author
-    AuthorAndPrefix(AuthorId, Vec<u8>),
+/// A query builder for document queries.
+#[derive(Debug, Default)]
+pub struct QueryBuilder<K> {
```

Comment:
we should move the whole query stuff into its own file

Reply:
Done

Reply:
I think you only moved the query stuff in the `fs` piece, not here